java/pgjdbc_hikaricp_using_dsql_connector/README.md

```diff
@@ -0,0 +1,159 @@
+
+Unlike the direct pgJDBC approach that requires custom token generation logic, the Aurora DSQL JDBC Connector handles all authentication complexity automatically while providing optimal performance with HikariCP.
+
+## ⚠️ Important
+
+- Running this code might result in charges to your AWS account.
+- We recommend that you grant your code the least privilege. At most, grant only the
+  minimum permissions required to perform the task. For more information, see
+  [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege).
+- This code is not tested in every AWS Region. For more information, see
+  [AWS Regional Services](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services).
+
+## Run the example
+
+### Prerequisites
+
+- You must have an AWS account, and have your default credentials and AWS Region
+  configured as described in the
+  [Globally configuring AWS SDKs and tools](https://docs.aws.amazon.com/credref/latest/refdocs/creds-config-files.html)
+  guide.
+- Java Development Kit (JDK): Ensure you have JDK 17+ installed.
+  _To verify that Java is installed, you can run:_
+  ```bash
+  java -version
+  ```
+- Build Tool: Gradle is used for this project.
+  - The project includes a Gradle connector, so no separate Gradle installation is required.
+- Aurora DSQL JDBC Connector: The project depends on the Aurora DSQL JDBC Connector library.
+
+- You must have an Aurora DSQL cluster. For information about creating an Aurora DSQL cluster, see the
+  [Getting started with Aurora DSQL](https://docs.aws.amazon.com/aurora-dsql/latest/userguide/getting-started.html)
+  guide.
+- If connecting as a non-admin user, ensure the user is linked to an IAM role and is granted access to the appropriate
+  schema. See the
+  [Using database roles with IAM roles](https://docs.aws.amazon.com/aurora-dsql/latest/userguide/using-database-and-iam-roles.html)
+  guide.
+
+The example demonstrates the following operations:
+
+- Creating a pool of connections to an Aurora DSQL cluster using the Aurora DSQL JDBC Connector
+- Creating a table
+- Inserting and querying data from different connections from the pool
+- Automatic token generation handled by the connector
+
+The example is designed to work with both admin and non-admin users:
+- When run as an admin user, it uses the `public` schema
+- When run as a non-admin user, it uses the `myschema` schema
+
+### Environment Variables
+
+Set the following environment variables for your cluster details:
+
+```bash
+# Required: Aurora DSQL cluster user (e.g., "admin")
+export CLUSTER_USER="<your-user>"
+
+# Option 1: Full Aurora DSQL endpoint
+export CLUSTER_ENDPOINT="<your-endpoint>"
+# e.g., "foo0bar1baz2quux3quuux4.dsql.us-east-1.on.aws"
+```
+
+### Run the code
+
+Build and run the example:
+
+```bash
+# Build the project
+./gradlew build
```

Comment:
Similar to other sample, there is no Gradle wrapper included in this sample so these build steps won't work.

We should also remove the `setup-env.sh` script.